README
NAME

    Dist::Zilla::Role::PluginBundle::Zilla - adds the zilla object and the
    logger to your bundles

VERSION

    Version 0.001

SYNOPSIS

 Dist::Zilla::Role::PluginBundle

        package Dist::Zilla::PluginBundle::MyBundle;

        use Moose;
        with 'Dist::Zilla::Role::PluginBundle::Zilla';

        sub bundle_config {
            my ( $class, $section ) = @_;

            my $self = $class->new($section);
            $self->log('Hello from your friendly bundle! We are running in ' . $self->zilla->root);
            $self->log_fatal('Something went wrong...');

            return;
        }

 Dist::Zilla::Role::PluginBundle::Easy

        package Dist::Zilla::PluginBundle::MyBundle;

        use Moose;
        with 'Dist::Zilla::Role::PluginBundle::Easy';
        with 'Dist::Zilla::Role::PluginBundle::Zilla';

        sub configure {
            my ($self) = @_;

            $self->log('Hello from your friendly bundle! We are running in ' . $self->zilla->root);
            $self->log_fatal('Something went wrong...');

            return;
        }

DESCRIPTION

    This role makes the zilla object available and adds the log, log_fatal,
    and log_debug methods to your bundle.

    This allows you to use the same logging procedure in your bundle that
    plugins use.

    If you use Dist::Zilla::Role::PluginBundle::Easy you have to import
    these two roles with two separate calls to with because both roles
    declare the name attribute.

SUPPORT

 Bugs / Feature Requests

    Please report any bugs or feature requests through the issue tracker at
    https://github.com/skirmess/Dist-Zilla-Role-PluginBundle-Zilla/issues.
    You will be notified automatically of any progress on your issue.

 Source Code

    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    https://github.com/skirmess/Dist-Zilla-Role-PluginBundle-Zilla

      git clone https://github.com/skirmess/Dist-Zilla-Role-PluginBundle-Zilla.git

AUTHOR

    Sven Kirmess <sven.kirmess@kzone.ch>

SEE ALSO

    Dist::Zilla::Role::PluginBundle, Dist::Zilla::Role::PluginBundle::Easy



Changes
Revision history for Dist-Zilla-Role-PluginBundle-Zilla

{{$NEXT}}

  - Switch license to the ISC license

0.001  2020-08-06 19:07:42 UTC

  - Initial release


t/lib/Local/PluginBundle.pm
# vim: ts=4 sts=4 sw=4 et: syntax=perl
#
# Copyright (c) 2020-2022 Sven Kirmess
#
# Permission to use, copy, modify, and distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

use 5.006;
use strict;
use warnings;

package Local::PluginBundle;

our $VERSION = '0.001';

use Moose;
with 'Dist::Zilla::Role::PluginBundle::Zilla';

use namespace::autoclean;

sub bundle_config {
    my ( $class, $section ) = @_;

    my $self = $class->new($section);

    $self->log( 'Hello from ' . __PACKAGE__ . ': name => ' . $self->name );

    die 'Zilla is not a Dist::Zilla' if !$self->zilla->isa('Dist::Zilla');    ## no critic (ErrorHandling::RequireCarping)

    return;
}

__PACKAGE__->meta->make_immutable;

1;


t/lib/Local/PluginBundleEasy.pm
# vim: ts=4 sts=4 sw=4 et: syntax=perl
#
# Copyright (c) 2020-2022 Sven Kirmess
#
# Permission to use, copy, modify, and distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

use 5.006;
use strict;
use warnings;

package Local::PluginBundleEasy;

our $VERSION = '0.001';

use Moose;
with 'Dist::Zilla::Role::PluginBundle::Easy';
with 'Dist::Zilla::Role::PluginBundle::Zilla';    ## no critic (Moose::ProhibitMultipleWiths)

use namespace::autoclean;

sub configure {
    my ($self) = @_;

    $self->log( 'Hello from ' . __PACKAGE__ . ': name => ' . $self->name );

    die 'Zilla is not a Dist::Zilla' if !$self->zilla->isa('Dist::Zilla');    ## no critic (ErrorHandling::RequireCarping)

    return;
}

__PACKAGE__->meta->make_immutable;

1;


Makefile.PL
use 5.006;
use strict;
use warnings;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
    ABSTRACT           => 'adds the zilla object and the logger to your bundles',
    AUTHOR             => 'Sven Kirmess <sven.kirmess@kzone.ch>',
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => 0,
    },
    DISTNAME         => 'Dist-Zilla-Role-PluginBundle-Zilla',
    LICENSE          => 'open_source',
    MIN_PERL_VERSION => '5.006',
    NAME             => 'Dist::Zilla::Role::PluginBundle::Zilla',
    PREREQ_PM        => {
        'Dist::Zilla::Role::PluginBundle' => 0,
        'List::Util'                      => 0,
        'Moose::Role'                     => 0,
        'Moose::Util::TypeConstraints'    => 0,
        'Scalar::Util'                    => 0,
        'namespace::autoclean'            => 0,
        strict                            => 0,
        warnings                          => 0,
    },
    TEST_REQUIRES => {
        Carp                                    => 0,
        Cwd                                     => 0,
        'Dist::Zilla::Role::PluginBundle::Easy' => 0,
        Exporter                                => 0,
        'File::Basename'                        => 0,
        'File::Path'                            => '2.07',
        'File::Spec'                            => 0,
        'File::Spec::Functions'                 => 0,
        Moose                                   => 0,
        'Test::DZil'                            => 0,
        'Test::More'                            => '0.88',
        lib                                     => 0,
    },
    VERSION => '0.001',
    test    => {
        TESTS => 't/*.t',
    },
);

my %FallbackPrereqs = (
    Carp                                    => 0,
    Cwd                                     => 0,
    'Dist::Zilla::Role::PluginBundle'       => 0,
    'Dist::Zilla::Role::PluginBundle::Easy' => 0,
    Exporter                                => 0,
    'File::Basename'                        => 0,
    'File::Path'                            => '2.07',
    'File::Spec'                            => 0,
    'File::Spec::Functions'                 => 0,
    'List::Util'                            => 0,
    Moose                                   => 0,
    'Moose::Role'                           => 0,
    'Moose::Util::TypeConstraints'          => 0,
    'Scalar::Util'                          => 0,
    'Test::DZil'                            => 0,
    'Test::More'                            => '0.88',
    lib                                     => 0,
    'namespace::autoclean'                  => 0,
    strict                                  => 0,
    warnings                                => 0,
);

if ( !eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
    delete $WriteMakefileArgs{TEST_REQUIRES};
    delete $WriteMakefileArgs{BUILD_REQUIRES};
    $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

if ( !eval { ExtUtils::MakeMaker->VERSION(6.52) } ) {
    delete $WriteMakefileArgs{CONFIGURE_REQUIRES};
}

WriteMakefile(%WriteMakefileArgs);

# vim: ts=4 sts=4 sw=4 et: syntax=perl
